Mathlib/Algebra/Category/ModuleCat/Adjunctions.lean

```diff
@@ -42,8 +42,14 @@ def free : Type u ⥤ ModuleCat R where
 
 variable {R}
 
+@[simp]
+theorem carrier_obj_free {X : Type u} : ((free R).obj X).carrier = (X →₀ R) := rfl
+@[simp]
+theorem hom_map_free {X Y : Type u} (f : X → Y) :
+    ((free R).map f).hom = Finsupp.lmapDomain _ _ f := rfl
+
 /-- Constructor for elements in the module `(free R).obj X`. -/
-noncomputable def freeMk {X : Type u} (x : X) : (free R).obj X := Finsupp.single x 1
+noncomputable abbrev freeMk {X : Type u} (x : X) : (free R).obj X := Finsupp.single x 1
```

Comment:
The use of `def` was a layer of isolation between `free` and `Finsupp.single`  which was made on purpose. I would not recommend trying to change this.

Reply:
The issue here is that there is no casting function to key on when converting between the two, and so this is an abstraction later that can never be unfolded automatically.

Reply:
Andrew Yang (standing next to me) has suggested that we do want lemmas like
```
@[simp]
theorem carrier_tensorObj {M N : ModuleCat.{u} R} :
    (M ⊗ N).carrier = TensorProduct R M.carrier N.carrier :=
  rfl
```
to be simp globally, but a temporary fix would be to make them local simp lemmas if that's what Joel thinks is best for now. The changes in this PR look very positive to me (but I'm certainly no expert in simp and normal forms etc).

Reply:
I think I've got this working with the `@[simp]`s removed from `carrier_obj_free, hom_map_free` and `freeMk` made a def again. This should make the diff smaller and if we do end up wanting `carrier_obj_free, hom_map_free` to be simp lemmas maybe that should be left to a different PR